src/Common/src/Internal/Runtime/TypeManagerHandle.cs

```diff
@@ -30,39 +36,38 @@ public IntPtr GetIntPtrUNSAFE()
             return _handleValue;
         }
 
-        public override int GetHashCode()
-        {
-            return _handleValue.GetHashCode();
-        }
-
-        public override bool Equals(object o)
-        {
-            if (!(o is TypeManagerHandle))
-                return false;
-
-            return _handleValue == ((TypeManagerHandle)o)._handleValue;
-        }
-
-        public static bool operator ==(TypeManagerHandle left, TypeManagerHandle right)
+        public bool IsNull
         {
-            return left._handleValue == right._handleValue;
+            get
+            {
+                return _handleValue == IntPtr.Zero;
+            }
         }
 
-        public static bool operator !=(TypeManagerHandle left, TypeManagerHandle right)
+        private unsafe TypeManager* AsTypeManagerPtr
         {
-            return left._handleValue != right._handleValue;
+            get
+            {
+                unsafe
+                {
+                    return (TypeManager*)(((byte*)(void*)_handleValue) - 1);
```

Comment:
The low-bit flag is left-over from ProjectN. I would be nice to simplify this.

Reply:
Yeah, deleting that and the `EETYPE_TYPE_MANAGER` cruft is on my list. I'll do it in a separate pull request.